app/demo-opentitan-earlgrey/src/main.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+extern crate panic_halt;
+extern crate riscv_rt;
+
+use riscv_rt::entry;
+
+#[entry]
+fn main() -> ! {
+    // The opentitan requires enabling the timer and interrupts
+    // for mtime and mtimecmp to behave as expected
+    // See https://docs.opentitan.org/hw/ip/rv_timer/doc/
+    const OT_TIMER_CTRL: u32 = 0x4010_0004;
```

Comment:
Shouldn't these be provided as services by some driver or kernel instead of having the task do this on its own by accessing the raw addresses?

Reply:
The base address for the OT can be retrieved now through `task_peripherals_str()` in `build-util`.

Reply:
I think they could eventually get replaced to a call of something like `start_opentitan_timer()` but we currently don't have an opentitan crate to put this in.  
The app main seems like the best place to put something that needs to run before the kernel.  Where in a task do you think it should go?

Reply:
> I think they could eventually get replaced to a call of something like start_opentitan_timer() but we currently don't have an opentitan crate to put this in.

That sounds good to me.

> The app main seems like the best place to put something that needs to run before the kernel. Where in a task do you think it should go?

I missed the fact that this is run as part of the kernel before the `start_kernel` call. Makes sense to keep it here. Like you mentioned above once we have an opentitan crate this can end up there.